File: src/main/java/com/mobilesolutionworks/android/zero/EntityLocator.java
package com.mobilesolutionworks.android.zero;

import android.support.annotation.NonNull;

/**
 * Created by yunarta on 7/5/17.
 */

public abstract class EntityLocator<T extends EntityLocator.Contract> {

    abstract T getInterface();

    private Struts.BackEnd struts;

    void setup(@NonNull Struts.BackEnd struts) {
        this.struts = struts;
    }

    /**
     * Created by yunarta on 9/5/17.
     */

    interface Contract {

    }
}

File: src/main/kotlin/com/mobilesolutionworks/android/struts/EntityLocator.kt
package com.mobilesolutionworks.android.struts

/**
 * Created by yunarta on 13/5/17.
 */

abstract class EntityLocator<out T : EntityLocator.Contract> {

    internal val _contract: T
        get() = contract

    protected abstract val contract: T

//    protected var struts: ImmutableStruts? = null
//        private set
//
//    internal fun setup(struts: ImmutableStruts) {
//        this.struts = struts
//    }

    /**
     * Created by yunarta on 9/5/17.
     */

    interface Contract
}